packages/gatsby-cli/src/issue/telemetry.js

```diff
@@ -0,0 +1,81 @@
+const path = require(`path`)
+const fs = require(`fs-extra`)
+
+const envInfo = require(`../env-info.js`)
+
+const mdCodeBlock = (content, lang) =>
+  `
+\`\`\`${lang}
+${content}
+\`\`\`
+  `.trim()
+
+module.exports = answers => {
+  if (!answers.envinfoConfirmation && !answers.filesConfirmation) {
+    return new Promise(resolve => resolve(``))
+  }
+
+  const promises = []
+
+  if (answers.envinfoConfirmation) {
+    promises.push(
+      envInfo({ markdown: true }).then((result, error) =>
+        `
+<details>
+<summary>Environment</summary>
+
+${(result || error).toString().trim()}
+
+</details>
+        `.trim()
+      )
+    )
+  }
+
+  if (answers.filesConfirmation) {
+    const files = [
+      `gatsby-browser.js`,
+      `gatsby-config.js`,
+      `gatsby-node.js`,
+      `gatsby-ssr.js`,
+    ]
+
+    files.forEach(fileName => {
+      promises.push(
+        fs
+          .readFile(path.join(process.cwd(), fileName))
+          .then((result, error) => {
```

Comment:
The error argument passed to the then handler is a callback function containing the error, not a simple value. See [this MDN document](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/then#Syntax)

I think it's probably a bit cleaner to handle the success case (with a `then`) and the failure case with a `.catch`. It will also simplify the `const content = ` creation there, since that's a bit complex to parse. 

What do you think?

Reply:
(also this was used once elsewhere at least, so take another look at 
any `.then((value, error)` patterns, as I'd recommend refactoring to explicit `.then(value)` and `.catch(error)` patterns.

Reply:
Oops, not sure how i missed that when testing :sweat_smile: 

I will use `.then`/`.catch`, thanks.